src/components/LibraryItem.tsx

```diff
@@ -184,9 +185,6 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
     }
 
     onLibraryItemClicked() {
-
-        // Toggle expansion state.
-        let currentlyExpanded = this.state.expanded;
-        this.setState({ expanded: !currentlyExpanded });
+        LibraryEntryPoint.dispatchEvent("libraryItemClick", this);
```

Comment:
Okay here's how the final code should look like:

```ts
onLibraryItemClicked() {
    let currentlyExpanded = this.state.expanded;
    this.setState({ expanded: !currentlyExpanded });

    let libraryView = this.props.libraryView;
    libraryView.raiseEvent("itemClicked", this. data.creationName);
}
```

And in the `LibraryView`, the code should look this way:

```ts
export class LibraryView {

    reactor: Reactor = null;

    constructor() {
        this.reactor = new Reactor();
    }

    on(eventName: string, callback: Function) {
        this.reactor.registerEvent(eventName, callback);
    }

    raiseEvent(params?: any | any[]): void {
        this.reactor.raiseEvent(params);
    }
}
```

Then how an HTML site registers a callback is like this:

```ts
let libraryView = ...;
libraryView.on("itemClicked", function(creationName) {
    console.log(creationName); // This should output creation name of the clicked item.
});
```